src/Databases/DatabaseAtomic.cpp

```diff
@@ -160,6 +160,46 @@ void DatabaseAtomic::dropTableImpl(ContextPtr local_context, const String & tabl
     DatabaseCatalog::instance().enqueueDroppedTableCleanup(table->getStorageID(), table, table_metadata_path_drop, sync);
 }
 
+void DatabaseAtomic::dropDetachedTable(ContextPtr local_context, const String & table_name, const bool sync)
+{
+    waitDatabaseStarted();
+
+    assert(fs::exists(getObjectMetadataPath(table_name)));
+
+    const String table_metadata_path = getObjectMetadataPath(table_name);
+    const UUID uuid_table = GetTableUUIDFromDetachedMetadata(local_context, table_metadata_path);
+
+    waitDetachedTableNotInUse(uuid_table);
+
+    const StorageID storage_id{getDatabaseName(), table_name, uuid_table};
+    const String table_metadata_path_drop = DatabaseCatalog::instance().getPathForDroppedMetadata(storage_id);
+
+    {
+        std::lock_guard lock(mutex);
+
+        auto txn = local_context->getZooKeeperMetadataTransaction();
+        if (txn && !local_context->isInternalSubquery())
+            txn->commit();
+
+        fs::rename(table_metadata_path, table_metadata_path_drop);
+        LOG_TRACE(log, "Rename {} to {} for removing.", table_metadata_path, table_metadata_path_drop);
+
+        DatabaseCatalog::instance().removeUUIDMappingFinally(uuid_table);
```

Comment:
It's too early, should be removed in DatabaseCatalog::dropTableFinally

Reply:
Ok, Im removing it from here.